src/System.Windows.Forms/src/System/Windows/Forms/Controls/PropertyGrid/PropertyGrid.cs

```
@@ -1987,8 +1987,13 @@ private static void DrawXorBar(Control targetControl, Rectangle rcFrame)
         }
     }
 
-    private bool EnablePropPageButton(object obj)
+    private bool EnablePropPageButton(object? obj)
     {
+        if (_viewPropertyPagesButton is null)
+        {
+            throw new InvalidOperationException();
+        }
+
```

why don't you add `!` below

---

This is desired. If we cannot gauruntee that the variable is non null by reasoning with the current code or refactor the code so that the variable is non null at the point of use, we should throw either `InvalidOperationException` or some argument exception to avoid a null reference exception.